lib/board.rb

```diff
@@ -0,0 +1,61 @@
+class Board
+	attr_accessor :cells
+
+	def initialize(board = Array.new(9, " "))
+		@cells = board
+		# @cells = Array.new(9, " ")
+	end
+
+	def reset!
+		self.cells.clear
+		self.cells = Array.new(9, " ")
+	end
+
+	def display
+		puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+		puts "-----------"
+		puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+		puts "-----------"
+		puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+	end
+
+	def position(user_input)
+		self.cells[user_input.to_i-1] #returns player from user input
+	end
+
+	def full?
+		self.cells.each do |cell| 
+			if cell == " " 
+				return false
+			end
+		end
+		true
+	end
+
+	def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```